lib/client/rpc/proposals.ts

```diff
@@ -242,7 +251,184 @@ export class FutarchyRPCProposalsClient implements FutarchyProposalsClient {
         ),
       ])
       .instruction();
-
     return ix;
   }
+  createMemoInstruction() {
+    return {
+      programId: new PublicKey("MemoSq4gqABAXKb96qnH8TysNcWxMyWCqXgDLGmfcHr"),
+      accounts: [],
+      data: Buffer.from("Memo", "utf8")
+    }
+  }
+
+  async simulate(tx: Transaction) {
+    const { blockhash } = await this.rpcProvider.connection.getLatestBlockhash();
+    tx.recentBlockhash = blockhash;
+    tx.feePayer = this.rpcProvider.wallet.publicKey; // Assuming the provider has a wallet
+
+    try {
+
+      const simulationResult = this.rpcProvider.connection.simulateTransaction(tx)
+      console.log(simulationResult)
+    } catch (e) {
+      console.error("Simulation error:", e)
+    }
+    // const wireTransaction = tx.
+  }
+
+  async getUserTokenBalance(tokenMintAddress: PublicKey) {
+    // Get the associated token account for the user's wallet and the token mint
+    const userTokenAccountAddress = await getAssociatedTokenAddress(
+      new PublicKey(tokenMintAddress),
+      this.rpcProvider.publicKey
+    );
+
+    // Fetch the account info
+    const accountInfo = await getAccount(this.rpcProvider.connection, userTokenAccountAddress);
+
+    // The amount is a BigInt, convert it to a string if necessary
+    return accountInfo.amount.toString();
+  }
+
+  async mintIx(amount: number,
+    vaultAccountAddress: PublicKey,
+    vaultAccount: VaultAccountWithProtocol) {
+    const ixs = [
+      createAssociatedTokenAccountIdempotentInstruction(
```

Comment:
So this is one of those fun ones (open to suggestions too). We call this for every mintIx, when really once we know those accounts exist (or should exist) we don't always need to inject them into the transaction.

So there's some world where we actually want to do a check if account exists (and holds rent or has been initialized, not just balance check) and then if it DOES NOT exist we inject these instructions... **But just a note more than anything**, as I don't have an elegant solution for it at this time.

Reply:
It's less about CUs and more about byte limit per txn...